### PR TITLE
Do less work inside lock scope and reduce memory allocation call

### PR DESCRIPTION
**Pull request by ngaut**

### CODE REVIEW

File: level_handler.go [hunk 1]
```diff
@@ -213,9 +223,9 @@ func (s *levelHandler) close() error {
 }
 
 // getTableForKey acquires a read-lock to access s.tables. It returns a list of tableHandlers.
-func (s *levelHandler) getTableForKey(key []byte) ([]*table.Table, func() error) {
+func (s *levelHandler) getTableForKey(key []byte) []*table.Table {
 	s.RLock()
-	defer s.RUnlock()
+	//	defer s.RUnlock()
```
Comment (coocood) [inline level_handler.go]: I think we better use defer here.

File: level_handler.go [hunk 2]
```diff
@@ -226,44 +236,48 @@ func (s *levelHandler) getTableForKey(key []byte) ([]*table.Table, func() error)
...
+		}
+	}()
+
 	keyNoTs := y.ParseKey(key)
 
 	var maxVs y.ValueStruct
 	for _, th := range tables {
 		if th.DoesNotHave(keyNoTs) {
-			y.NumLSMBloomHits.Add(s.strLevel, 1)
+			//y.NumLSMBloomHits.Add(s.strLevel, 1)
```
Comment (coocood) [inline level_handler.go]: We don't need the stats?

File: db.go [hunk 3]
```diff
@@ -464,11 +464,7 @@ func (db *DB) getMemTables() ([]*skl.Skiplist, func()) {
 		immt.IncrRef()
 		tables = append(tables, immt)
 	}
-	return tables, func() {
-		for _, tbl := range tables {
-			tbl.DecrRef()
-		}
-	}
+	return tables
```
Comment (coocood) [inline db.go]: This change seems don't reduce lock scope.

Comment (ngaut) [reply db.go]: It return tables explicitly, let the caller to do clean up staff.

File: level_handler.go (hunk 2 above)
Comment (ngaut) [reply level_handler.go]: We will redesign the stats show.

Comment (coocood) [reply level_handler.go]: Can we just delete the line instead of comment out?

**Pull request merged**